repo/common/common_test.go

```diff
@@ -0,0 +1,132 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/ipfs/go-ipfs/thirdparty/assert"
+)
+
+func TestMapMergeDeepReturnsNew(t *testing.T) {
+	leftMap := make(map[string]interface{})
+	leftMap["A"] = "Hello World"
+
+	rightMap := make(map[string]interface{})
+	rightMap["A"] = "Foo"
+
+	MapMergeDeep(leftMap, rightMap)
+
+	assert.True(leftMap["A"] == "Hello World", t, "MapMergeDeep should return a new map instance")
+}
+
+func TestMapMergeDeepNewKey(t *testing.T) {
+	leftMap := make(map[string]interface{})
+	leftMap["A"] = "Hello World"
+	/*
+		leftMap
+		{
+			A: "Hello World"
+		}
+	*/
+
+	rightMap := make(map[string]interface{})
+	rightMap["B"] = "Bar"
+	/*
+		rightMap
+		{
+			B: "Bar"
+		}
+	*/
+
+	result := MapMergeDeep(leftMap, rightMap)
+	/*
+		expected
+		{
+			A: "Hello World"
+			B: "Bar"
+		}
+	*/
+
+	assert.True(result["B"] == "Bar", t, "New keys in right map should exist in resulting map")
+}
+
+func TestMapMergeDeepRecursesOnMaps(t *testing.T) {
+	leftMapA := make(map[string]interface{})
+	leftMapA["B"] = "A value!"
+	leftMapA["C"] = "Another value!"
+
+	leftMap := make(map[string]interface{})
+	leftMap["A"] = leftMapA
+	/*
+		leftMap
+		{
+			A: {
+				B: "A value!"
+				C: "Another value!"
+			}
+		}
+	*/
+
+	rightMapA := make(map[string]interface{})
+	rightMapA["C"] = "A different value!"
+
+	rightMap := make(map[string]interface{})
+	rightMap["A"] = rightMapA
+	/*
+		rightMap
+		{
+			A: {
+				C: "A different value!"
+			}
+		}
+	*/
+
+	result := MapMergeDeep(leftMap, rightMap)
+	/*
+		expected
+		{
+			A: {
+				B: "A value!"
+				C: "A different value!"
+			}
+		}
+	*/
```

Comment:
We should be comparing the entire maps with `require.Equal`, that way the comment  actually becomes the test, and we don't need specific equality checks.